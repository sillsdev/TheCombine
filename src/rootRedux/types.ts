import { type Action } from "redux";
import { type ThunkDispatch } from "redux-thunk";

import {
  type LoginState,
  defaultState as loginState,
} from "components/Login/Redux/LoginReduxTypes";
import {
  type CurrentProjectState,
  defaultState as currentProjectState,
} from "components/Project/ProjectReduxTypes";
import {
  type ExportProjectState,
  defaultState as exportProjectState,
} from "components/ProjectExport/Redux/ExportProjectReduxTypes";
import {
  type PronunciationsState,
  defaultState as pronunciationsState,
} from "components/Pronunciations/Redux/PronunciationsReduxTypes";
import {
  type TreeViewState,
  defaultState as treeViewState,
} from "components/TreeView/Redux/TreeViewReduxTypes";
import {
  type CharacterInventoryState,
  defaultState as characterInventoryState,
} from "goals/CharacterInventory/Redux/CharacterInventoryReduxTypes";
import {
  type MergeTreeState,
  defaultState as mergeDuplicateGoal,
} from "goals/MergeDuplicates/Redux/MergeDupsReduxTypes";
import {
  type GoalsState,
  defaultState as goalsState,
} from "goals/Redux/GoalReduxTypes";
import {
  type AnalyticsState,
  defaultState as analyticsState,
} from "types/Redux/analyticsReduxTypes";

/** Redux store structure */
export interface StoreState {
  //login
  loginState: LoginState;

  //project
  currentProjectState: CurrentProjectState;
  exportProjectState: ExportProjectState;

  //data entry and review entries
  treeViewState: TreeViewState;
  pronunciationsState: PronunciationsState;

  //goal timeline and current goal
  goalsState: GoalsState;

  //merge duplicates goal
  mergeDuplicateGoal: MergeTreeState;

  //character inventory goal
  characterInventoryState: CharacterInventoryState;

  //analytics state
  analyticsState: AnalyticsState;
}

/** Default values for the Redux store */
export const defaultState: StoreState = {
  loginState: { ...loginState },
  currentProjectState: { ...currentProjectState },
  exportProjectState: { ...exportProjectState },
  treeViewState: { ...treeViewState },
  pronunciationsState: { ...pronunciationsState },
  goalsState: { ...goalsState },
  mergeDuplicateGoal: { ...mergeDuplicateGoal },
  characterInventoryState: { ...characterInventoryState },
  analyticsState: { ...analyticsState },
};

export interface ActionWithPayload<T> extends Action {
  payload: T;
}

// https://redux.js.org/usage/usage-with-typescript#type-checking-redux-thunks
// suggests a custom general type for ThunkAction,
// so in like fashion, here's one for ThunkDispatch:
export type StoreStateDispatch = ThunkDispatch<StoreState, any, Action>;
