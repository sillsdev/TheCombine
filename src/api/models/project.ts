/* tslint:disable */
/* eslint-disable */
/**
 * BackendFramework
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { AutocompleteSetting } from "./autocomplete-setting";
import { CustomField } from "./custom-field";
import { EmailInvite } from "./email-invite";
import { SemanticDomain } from "./semantic-domain";
import { WritingSystem } from "./writing-system";

/**
 *
 * @export
 * @interface Project
 */
export interface Project {
  /**
   *
   * @type {string}
   * @memberof Project
   */
  id: string;
  /**
   *
   * @type {string}
   * @memberof Project
   */
  name: string;
  /**
   *
   * @type {boolean}
   * @memberof Project
   */
  isActive: boolean;
  /**
   *
   * @type {boolean}
   * @memberof Project
   */
  liftImported: boolean;
  /**
   *
   * @type {boolean}
   * @memberof Project
   */
  definitionsEnabled: boolean;
  /**
   *
   * @type {WritingSystem}
   * @memberof Project
   */
  semDomWritingSystem: WritingSystem;
  /**
   *
   * @type {AutocompleteSetting}
   * @memberof Project
   */
  autocompleteSetting: AutocompleteSetting;
  /**
   *
   * @type {WritingSystem}
   * @memberof Project
   */
  vernacularWritingSystem: WritingSystem;
  /**
   *
   * @type {Array<WritingSystem>}
   * @memberof Project
   */
  analysisWritingSystems: Array<WritingSystem>;
  /**
   *
   * @type {Array<SemanticDomain>}
   * @memberof Project
   */
  semanticDomains: Array<SemanticDomain>;
  /**
   *
   * @type {Array<string>}
   * @memberof Project
   */
  validCharacters: Array<string>;
  /**
   *
   * @type {Array<string>}
   * @memberof Project
   */
  rejectedCharacters: Array<string>;
  /**
   *
   * @type {Array<CustomField>}
   * @memberof Project
   */
  customFields?: Array<CustomField> | null;
  /**
   *
   * @type {Array<string>}
   * @memberof Project
   */
  wordFields?: Array<string> | null;
  /**
   *
   * @type {Array<string>}
   * @memberof Project
   */
  partsOfSpeech?: Array<string> | null;
  /**
   *
   * @type {Array<EmailInvite>}
   * @memberof Project
   */
  inviteTokens: Array<EmailInvite>;
}
