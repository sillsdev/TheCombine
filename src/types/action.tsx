import { Action } from "redux";
import { Goal } from "./goals";
import { CreateCharInv } from "../goals/CreateCharInv/CreateCharInv";

export interface ActionWithPayload<T> extends Action {
  payload: T;
}

const MOCK_TYPE = "MOCK_TYPE";
type MOCK_TYPE = typeof MOCK_TYPE;

interface MockAction extends ActionWithPayload<Goal> {
  type: MOCK_TYPE;
  payload: Goal;
}

const goal: Goal = new CreateCharInv([]);

// Used for passing a non-existent action to a reducer
export const MockActionInstance: MockAction = {
  type: MOCK_TYPE,
  payload: goal
};
