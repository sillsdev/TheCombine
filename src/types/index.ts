import { LoginState } from "components/Login/Redux/LoginReduxTypes";
import { PasswordResetState } from "components/PasswordReset/Redux/ResetReduxTypes";
import { CurrentProjectState } from "components/Project/ProjectReduxTypes";
import { ExportProjectState } from "components/ProjectExport/Redux/ExportProjectReduxTypes";
import { CreateProjectState } from "components/ProjectScreen/CreateProject/Redux/CreateProjectReduxTypes";
import { PronunciationsState } from "components/Pronunciations/Redux/PronunciationsReduxTypes";
import { TreeViewState } from "components/TreeView/TreeViewReducer";
import { CharacterInventoryState } from "goals/CharInventoryCreation/Redux/CharacterInventoryReduxTypes";
import { MergeTreeState } from "goals/MergeDupGoal/Redux/MergeDupReduxTypes";
import { ReviewEntriesState } from "goals/ReviewEntries/ReviewEntriesComponent/Redux/ReviewEntriesReduxTypes";
import { AnalyticsState } from "types/Redux/analyticsReduxTypes";
import { GoalsState } from "types/goals";

//root store structure
export interface StoreState {
  //login
  loginState: LoginState;
  passwordResetState: PasswordResetState;

  //project
  createProjectState: CreateProjectState;
  currentProjectState: CurrentProjectState;
  exportProjectState: ExportProjectState;

  //data entry and review entries
  treeViewState: TreeViewState;
  reviewEntriesState: ReviewEntriesState;
  pronunciationsState: PronunciationsState;

  //goal timeline and current goal
  goalsState: GoalsState;

  //merge duplicates goal
  mergeDuplicateGoal: MergeTreeState;

  //character inventory goal
  characterInventoryState: CharacterInventoryState;

  //analytics state
  analyticsState: AnalyticsState;
}
