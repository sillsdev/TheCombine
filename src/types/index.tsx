import { ThunkAction, ThunkDispatch } from "redux-thunk";

import { TempState } from "../components/Temp/TempReducer";
import { TempAction } from "../components/Temp/TempActions";
import { LocalizeState } from "react-localize-redux";
import { LoginState } from "../components/Login/LoginReducer";
import { CreateProjectState } from "../components/CreateProject/CreateProjectReducer";
import { GoalsState } from "./goals";

//root store structure
export interface StoreState {
  localize: LocalizeState;
  tempState: TempState;
  loginState: LoginState;
  createProjectState: CreateProjectState;
  goalsState: GoalsState;
}

//root action type
export type TCActions = TempAction; // | otherActions;

//thunk types
export type TCThunkResult<R> = ThunkAction<R, StoreState, null, TCActions>;
export type TCThunkDispatch = ThunkDispatch<StoreState, null, TCActions>;
