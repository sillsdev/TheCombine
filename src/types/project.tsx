import { Word, SemanticDomain } from "./word";

export interface CustomField {
  name: string;
  type: string;
}

export interface Project {
  id: string;
  name: string;
  semanticDomains: SemanticDomain[];
  userRoles: string;
  vernacularWritingSystem: string;
  analysisWritingSystems: string[];
  characterSet: string[];
  wordFields: string[];
  partsOfSpeech: string[];
  words: Word[];
  customFields: CustomField[];
}

export const defaultProject = {
  id: "",
  name: "",
  semanticDomains: [],
  userRoles: "",
  vernacularWritingSystem: "",
  analysisWritingSystems: [],
  characterSet: [],
  customFields: [],
  wordFields: [],
  partsOfSpeech: [],
  words: []
} as Project;
