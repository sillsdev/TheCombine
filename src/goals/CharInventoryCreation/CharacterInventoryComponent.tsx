import * as React from "react";
import {
  LocalizeContextProps,
  withLocalize,
  Translate,
  TranslateFunction
} from "react-localize-redux";
import CharacterSet from "./components/CharacterSet";
import {
  Grid,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions
} from "@material-ui/core";
import { Project } from "../../types/project";
import SampleWords from "./components/SampleWords";
import { Save } from "@material-ui/icons";
import history from "../../history";

export interface CharacterInventoryProps {
  setInventory: (inventory: string[]) => void;
  uploadInventory: () => void;
  inventory: string[];
  currentProject: Project;
  translate: TranslateFunction;
}

interface CharacterInventoryState {
  cancelDialogOpen: boolean;
}

class CharacterInventory extends React.Component<
  CharacterInventoryProps & LocalizeContextProps,
  CharacterInventoryState
> {
  constructor(props: CharacterInventoryProps & LocalizeContextProps) {
    super(props);
    // Load inventory from server
    this.props.setInventory(this.props.currentProject.characterSet);
    this.state = { cancelDialogOpen: false };
  }

  handleClose() {
    this.setState({ cancelDialogOpen: false });
  }

  render() {
    return (
      <div>
        <Grid
          container
          justify="center"
          spacing={2}
          style={{ background: "#fff" }}
        >
          <Grid item sm={6} xs={12}>
            <CharacterSet
              setInventory={inventory => this.props.setInventory(inventory)}
              inventory={this.props.inventory}
            />
          </Grid>
          <Grid item sm={6} xs={12}>
            <SampleWords
              setInventory={inventory => this.props.setInventory(inventory)}
              inventory={this.props.inventory}
            />
          </Grid>
          <Grid item xs={12}>
            {/* submission buttons */}
            <Grid container justify="center">
              <Button
                variant="contained"
                color="primary"
                onClick={() => {
                  this.props.uploadInventory();
                  history.push("/goals");
                }}
                style={{ margin: 10 }} // remove when we can add theme
              >
                <Save /> <Translate id="charInventory.save" />
              </Button>
              <Button
                variant="contained"
                onClick={() => {
                  this.setState({ cancelDialogOpen: true });
                }}
                style={{ margin: 10 }} // remove when we can add theme
              >
                {" "}
                <Translate id="charInventory.cancel" />
              </Button>
            </Grid>
          </Grid>
        </Grid>

        {/* "Are you sure?" dialog for the cancel button */}
        <Dialog
          open={this.state.cancelDialogOpen}
          onClose={() => this.handleClose()}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">
            <Translate id="charInventory.dialog.title" />
          </DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              <Translate id="charInventory.dialog.content" />
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button
              onClick={() => history.push("/goals")}
              variant="contained"
              color="secondary"
              autoFocus
            >
              <Translate id="charInventory.dialog.yes" />
            </Button>
            <Button onClick={() => this.handleClose()} color="primary">
              <Translate id="charInventory.dialog.no" />
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withLocalize(CharacterInventory);
