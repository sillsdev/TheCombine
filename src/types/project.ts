import { AutocompleteSetting, Project, WritingSystem } from "api/models";
import { randomIntString } from "utilities";

export function newWritingSystem(
  bcp47 = "",
  name = "",
  font = ""
): WritingSystem {
  return { bcp47, name, font };
}

export function newProject(name = ""): Project {
  return {
    id: "",
    name,
    isActive: true,
    liftImported: false,
    definitionsEnabled: false,
    semanticDomains: [],
    semDomWritingSystem: newWritingSystem(),
    vernacularWritingSystem: newWritingSystem(),
    analysisWritingSystems: [newWritingSystem()],
    validCharacters: [],
    rejectedCharacters: [],
    inviteTokens: [],
    autocompleteSetting: AutocompleteSetting.On,
  };
}

// Randomize properties as needed for tests.
export function randomProject(): Project {
  const project = newProject(randomIntString());
  project.id = randomIntString();
  project.isActive = Math.random() < 0.5;
  return project;
}
