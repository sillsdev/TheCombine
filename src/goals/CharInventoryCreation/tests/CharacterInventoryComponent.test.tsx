import React, { ReactElement, ReactNode } from "react";
import CharacterInventoryComponent, {
  CharacterInventory,
  SAVE,
  CANCEL
} from "../CharacterInventoryComponent";
import ReactDOM from "react-dom";
import { Project } from "../../../types/project";
import renderer, { ReactTestRenderer } from "react-test-renderer";

// Constants
const SET_INV = jest.fn();
const UPLOAD_INV = jest.fn();

// Variables
var charMaster: ReactTestRenderer;
var charHandle: CharacterInventory;

// This mock bypasses the fact that react-test-renderer does not support portals, with no clean solution. This bypasses the whole issue
// by replacing the portal-creating object (the Dialog) with a lightweight, innocuous Material-Ui component with no such glitchy properties.
jest.mock("@material-ui/core", () => {
  const materialUiCore = jest.requireActual("@material-ui/core");
  return {
    ...materialUiCore,
    Dialog: materialUiCore.Container
  };
});

beforeAll(() => {
  renderer.act(() => {
    charMaster = renderer.create(
      <CharacterInventoryComponent
        validCharacters={["a"]}
        currentProject={{ validCharacters: ["a"] } as Project}
        setValidCharacters={SET_INV}
        uploadInventory={UPLOAD_INV}
        addToValidCharacters={jest.fn()}
        setRejectedCharacters={jest.fn()}
        rejectedCharacters={[]}
      />
    );
  });
  charHandle = charMaster.root.findByType(CharacterInventory).instance;
});

beforeEach(() => {
  SET_INV.mockClear();
  UPLOAD_INV.mockClear();
});

describe("Testing Character Inventory Component", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <CharacterInventoryComponent
        validCharacters={["a"]}
        currentProject={{ validCharacters: ["a"] } as Project}
        setValidCharacters={SET_INV}
        uploadInventory={UPLOAD_INV}
        addToValidCharacters={jest.fn()}
        setRejectedCharacters={jest.fn()}
        rejectedCharacters={[]}
      />,
      div
    );
    expect(ReactDOM.unmountComponentAtNode(div)).toBeTruthy();
  });

  it("Renders properly (snapshot test)", () => {
    expect(charMaster.toJSON()).toMatchSnapshot();
  });

  it("Attempts to save progress on save", () => {
    charMaster.root.findByProps({ id: SAVE }).props.onClick();

    expect(UPLOAD_INV).toHaveBeenCalledTimes(1);
  });

  it("Attempts to pop up a dialogue on cancel", () => {
    charMaster.root.findByProps({ id: CANCEL }).props.onClick();

    expect(charHandle.state.cancelDialogOpen).toBeTruthy();
  });

  it("Cancels dialog open on close", () => {
    charHandle.setState({ cancelDialogOpen: true });
    charHandle.handleClose();

    expect(charHandle.state.cancelDialogOpen).toBeFalsy();
  });
});
