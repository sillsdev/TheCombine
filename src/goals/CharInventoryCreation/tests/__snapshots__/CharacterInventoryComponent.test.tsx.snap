// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Character Inventory Component Renders properly (snapshot test) 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
    style={
      Object {
        "background": "#fff",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
      style={
        Object {
          "borderRight": "1px solid #ccc",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h1
            className="MuiTypography-root MuiTypography-h4"
          >
            Missing translationId: charInventory.characterSet.title for language: \${ languageCode }
             
            <svg
              aria-describedby={null}
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="presentation"
              title="Missing translationId: charInventory.characterSet.help for language: \${ languageCode }"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
              />
            </svg>
          </h1>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-1"
          onDragOver={[Function]}
          style={
            Object {
              "paddingBottom": 0,
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
          >
            <div
              draggable={true}
              id="charTile_a"
              onClick={[Function]}
              onDragEnd={[Function]}
              onDragStart={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "border": "4px solid #c8e6c9",
                  "borderRadius": "50%",
                  "cursor": "pointer",
                  "display": "flex",
                  "height": 40,
                  "justifyContent": "center",
                  "position": "relative",
                  "width": 40,
                }
              }
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                a
              </h6>
            </div>
            <h6
              className="MuiTypography-root MuiTypography-subtitle2"
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              Missing translationId: charInventory.characterSet.accepted for language: \${ languageCode }
            </h6>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            style={
              Object {
                "background": "whitesmoke",
                "borderTop": "1px solid #ccc",
                "padding": 16,
              }
            }
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                style={
                  Object {
                    "maxWidth": 512,
                  }
                }
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink={true}
                  htmlFor="valid-characters-input"
                >
                  Missing translationId: charInventory.characterSet.acceptedCharacters for language: \${ languageCode }
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-204 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-205"
                      style={
                        Object {
                          "width": 0,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <input
                    aria-invalid={false}
                    autoComplete="off"
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="valid-characters-input"
                    name="chracters"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    spellcheck="false"
                    style={
                      Object {
                        "letterSpacing": 5,
                      }
                    }
                    type="text"
                    value="a"
                  />
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style={
                Object {
                  "background": "whitesmoke",
                }
              }
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                style={
                  Object {
                    "maxWidth": 512,
                  }
                }
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                  htmlFor="rejected-characters-input"
                >
                  Missing translationId: charInventory.characterSet.rejectedCharacters for language: \${ languageCode }
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-204 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-205"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <input
                    aria-invalid={false}
                    autoComplete="off"
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="rejected-characters-input"
                    name="chracters"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    spellcheck="false"
                    style={
                      Object {
                        "letterSpacing": 5,
                      }
                    }
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h1
            className="MuiTypography-root MuiTypography-h5"
          >
            Missing translationId: charInventory.sampleWords.title for language: \${ languageCode }
          </h1>
          <h6
            className="MuiTypography-root MuiTypography-subtitle1"
          >
            Missing translationId: charInventory.sampleWords.description for language: \${ languageCode }
          </h6>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style={
        Object {
          "borderTop": "1px solid #ccc",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          id="pushGoals"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": 10,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
              />
            </svg>
             
            Missing translationId: charInventory.save for language: \${ languageCode }
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          id="cancelInventoryCreation"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": 10,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
             
            Missing translationId: charInventory.cancel for language: \${ languageCode }
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    className="MuiContainer-root MuiContainer-maxWidthLg"
    onClose={[Function]}
    open={false}
  >
    <div
      className="MuiDialogTitle-root"
      id="alert-dialog-title"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Missing translationId: charInventory.dialog.title for language: \${ languageCode }
      </h6>
    </div>
    <div
      className="MuiDialogContent-root"
    >
      <p
        className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        id="alert-dialog-description"
      >
        Missing translationId: charInventory.dialog.content for language: \${ languageCode }
      </p>
    </div>
    <div
      className="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <button
        autoFocus={true}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Missing translationId: charInventory.dialog.yes for language: \${ languageCode }
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Missing translationId: charInventory.dialog.no for language: \${ languageCode }
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
